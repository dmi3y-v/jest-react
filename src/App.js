import React, { useEffect, useState } from "react";

function App() {
  const [data, setData] = useState(null);
  useEffect(() => {
    setTimeout(() => {
      setData({});
    }, 100);
  }, []);
  return (
    <div>
      <h1>Hello world</h1>
      <button>click me</button>
      <input type="text" placeholder="input value..."></input>
      {data && <div>data</div>}
    </div>
  );
}

export default App;
