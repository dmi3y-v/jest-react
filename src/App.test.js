import { render, screen } from "@testing-library/react";
import App from "./App";

describe("TEST APP", () => {
  test("renders main App", () => {
    render(<App />);
    const helloWorldElement = screen.getByText(/hello world/i);
    const btn = screen.getByRole("button");
    const input = screen.getByPlaceholderText(/input value/i);
    expect(helloWorldElement).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });
  test("renders main App", async () => {
    render(<App />);
    // const helloWorldElement = screen.queryByText(/hello2/i);
    // expect(helloWorldElement).toBeNull();
    screen.debug();
    const helloWorldElement = await screen.findByText(/data/i);
    expect(helloWorldElement).toBeInTheDocument();
    screen.debug();
  });
});
