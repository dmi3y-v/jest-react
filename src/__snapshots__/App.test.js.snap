// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP renders main App 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
